jwst/ramp_fitting/tests/test_ramp_fit_step.py

```diff
@@ -0,0 +1,256 @@
+import pytest
+import numpy as np
+
+from jwst.ramp_fitting.ramp_fit_step import RampFitStep
+
+from jwst.datamodels import RampModel
+from jwst.datamodels import GainModel, ReadnoiseModel
+
+
+@pytest.fixture(scope="module")
+def generate_miri_reffiles(tmpdir_factory):
+    gainfile = str(tmpdir_factory.mktemp("data").join("gain.fits"))
+    readnoisefile = str(tmpdir_factory.mktemp("data").join('readnoise.fits'))
+
+    ingain = 6
+    xsize = 103
+    ysize = 102
+    gain = np.ones(shape=(ysize, xsize), dtype=np.float64) * ingain
+    gain_model = GainModel(data=gain)
+    gain_model.meta.instrument.name = "MIRI"
+    gain_model.meta.subarray.name = "FULL"
+    gain_model.meta.subarray.xstart = 1
+    gain_model.meta.subarray.ystart = 1
+    gain_model.meta.subarray.xsize = xsize
+    gain_model.meta.subarray.ysize = ysize
+    gain_model.save(gainfile)
+
+    inreadnoise = 5
+    rnoise = np.ones(shape=(ysize, xsize), dtype=np.float64) * inreadnoise
+    readnoise_model = ReadnoiseModel(data=rnoise)
+    readnoise_model.meta.instrument.name = "MIRI"
+    readnoise_model.meta.subarray.xstart = 1
+    readnoise_model.meta.subarray.ystart = 1
+    readnoise_model.meta.subarray.xsize = xsize
+    readnoise_model.meta.subarray.ysize = ysize
+    readnoise_model.save(readnoisefile)
+
+    return gainfile, readnoisefile
+
+
+@pytest.fixture
+def setup_inputs():
+
+    def _setup(ngroups=10, readnoise=10, nints=1, nrows=1024, ncols=1032,
+               nframes=1, grouptime=1.0, gain=1, deltatime=1):
+        times = np.array(list(range(ngroups)), dtype=np.float64) * deltatime
+        gain = np.ones(shape=(nrows, ncols), dtype=np.float64) * gain
+        err = np.ones(shape=(nints, ngroups, nrows, ncols), dtype=np.float64)
+        data = np.zeros(shape=(nints, ngroups, nrows, ncols), dtype=np.float64)
```

Comment:
Both of these arrays, as soon as they get assigned to `RampModel` will be coerced to `np.float32`, as specified in the `RampModel` schema.  So specifying them here as `np.float64` guarantees that an extra copy will happen a few lines down.

Further, since all these arrays (except `int_times`) are specified in the schema, there's not a need to pre-populate them here, especially if they will have arrays assigned to them a few lines down.  I.e. the default for `rampmodel.data` will be zeros if nothing is assigned to it.

Reply:
I changed the test to not use those two files in the creation of the `RampModel`.